csharp/src/Apache.Arrow/Arrays/ArrowArrayFactory.cs

```diff
@@ -25,6 +25,8 @@ public static IArrowArray BuildArray(ArrayData data)
         {
             switch (data.DataType.TypeId)
             {
+                case ArrowTypeId.Null:
```

Comment:
We should also update 

https://github.com/apache/arrow/blob/122fc86576b03ebd130e027200c5eda6a185d2ee/docs/source/status.rst?plain=1#L32-L35

for this new functionality.

We should also turn on this integration test as well:

https://github.com/apache/arrow/blob/122fc86576b03ebd130e027200c5eda6a185d2ee/dev/archery/archery/integration/datagen.py#L1672-L1673